### ISSUE TITLE
Flashing with avrdude

### ISSUE THREAD

**Issue by alexpirine** [sparkfun/OpenLog#219]
Hello,

I wanted to update the firmware of my OpenLog device, and I encountered a lot of problems on my Mac.

So I tried using `avrdude`. Sometimes flashing succeeds, sometimes it fails, but the device never properly works. And eventually, I can not even flash it anymore.

I compiled hex files in Arduino UNO (should I use Uno or Pro Mini as target?).

I then flashed them with `avrdude` (should I flash with or without bootloader ?).

I wonder if, before flashing, I should set the fuse values?

Currently, my chip reads:

```
avrdude: Device signature = 0x1e950f (probably m328p)
avrdude: safemode: lfuse reads as 62
avrdude: safemode: hfuse reads as D9
avrdude: safemode: efuse reads as FF
```

Should I change these values before flashing? I think my problems might be related to the fuse settings.

I found that values used by Arduino IDE are different:

https://github.com/arduino/Arduino/blob/dc5827e08b5e269d3d22322cefaed90a14191b89/hardware/arduino/avr/boards.txt

Which values should I use?

Thank you!

**Comment by alexpirine** [comment]
I think I bricked about 4 OpenLog devices now… I don't understand why writing OpenLog firmware to flash with OpenLog firmware changes fuses incorrectly sometimes. And sometimes it says that everything is OK, but when I boot the device, the TX/RX led is flashing and the device is not responding to any commands. Under oscilloscope the TX pin seem to send data at some non-standard baudrate…

The bricked devices no longer work and avrdude reads random chip identifiers now (like 0x00ff00, 0x0000ff…). If I try to force it with `-F`, it just says it cannot read fuses.

I am using an Arduino nano as ISP. I learned plenty of things, but still have so many random problems… I don't understand why is it so difficult to flash using the SPI pins? I never thought it I would struggle with this so much.
